internal/extender/resourcereservations.go

```diff
@@ -283,6 +386,27 @@ func (rrm *ResourceReservationManager) getActivePods(ctx context.Context, pod *v
 	return activePodNames, nil
 }
 
+func (rrm *ResourceReservationManager) onExecutorPodDeletion(obj interface{}) {
+	pod, ok := utils.GetPodFromObjectOrTombstone(obj)
+	if !ok {
+		rrm.logger.Warn("failed to parse object as pod, skipping")
+		return
+	}
+
+	// Only keep executor pods which are for dynamic allocation applications and which have resource reservations to give back
+	if _, ok = rrm.softReservationStore.GetSoftReservation(pod.Labels[common.SparkAppIDLabel]); ok {
+		if !rrm.softReservationStore.ExecutorHasSoftReservation(context.Background(), pod) {
```

Comment:
we can save the `ctx` instead of the `logger` and use it here, with the background context I believe logs will be dropped